Frameworks/Foundation/NSKVOSwizzling.mm

```diff
@@ -355,36 +367,45 @@ static SEL formatSelector(NSString* format, ...) {
 }
 
 // invariant: rawKey has already been capitalized
-static void _NSKVOEnsureCollectionWillNotify(id object, NSString* key, const char* rawKey) {
-    bool found = false;
-    auto insertOneSel = formatSelector(@"insertObject:in%sAtIndex:", rawKey);
-
-    if ([object respondsToSelector:insertOneSel]) {
-        found = true;
+static inline void _NSKVOEnsureOrderedCollectionWillNotify(id object, NSString* key, const char* rawKey) {
+    SEL insertOne = formatSelector(@"insertObject:in%sAtIndex:", rawKey);
+    SEL insertMany = formatSelector(@"insert%s:atIndexes:", rawKey);
+    if ([object respondsToSelector:insertOne] || [object respondsToSelector:insertMany]) {
```

Comment:
the point of the previous code was to _avoid even formatting_ a single selector that we wouldn't end up using